libmscore/keysig.cpp

```diff
@@ -129,18 +129,26 @@ void KeySig::layout()
       // AND style says they are not off
       // OR key sig is CMaj/Amin (in which case they are always shown)
 
-      bool naturalsOn = false;
-      Measure* prevMeas = measure() ? measure()->prevMeasure() : nullptr;
-
-      // If we're not force hiding naturals (Continuous panel), use score style settings
-      if (!_hideNaturals)
-          naturalsOn =
-            (prevMeas && prevMeas->sectionBreak() == nullptr
-            && (score()->styleI(StyleIdx::keySigNaturals) != int(KeySigNatural::NONE) || t1 == 0) );
-
-      // Don't repeat naturals if shown in courtesy
-      if (prevMeas && prevMeas->findSegment(Segment::Type::KeySigAnnounce, measure()->tick()) != 0
-          && segment()->segmentType() != Segment::Type::KeySigAnnounce )
+      Measure* m           = measure();
+      Measure* prevMeasure = m ? m->prevMeasure() : nullptr;
+
+      // display of naturals defaults according to style
+      bool naturalsOn = score()->styleI(StyleIdx::keySigNaturals) != int(KeySigNatural::NONE) || t1 == 0;
+      // suppress if we are hiding all naturals
+      if (_hideNaturals)
+            naturalsOn = false;
+      // suppress after section break
+      else if (prevMeasure && prevMeasure->isFinalMeasureOfSection())
```

Comment:
Just a guess - the problem in https://musescore.org/en/node/272691 may stem from this.  In theory, one would like to believe this code would work, but I could easily believe this flag might not have been initialized correctly in the case of mmrests especially immediately after a load  - we have experienced all sorts of problems where the underlying measures haven't been laid out yet.

It's possible that simply using m->prevMeasureMM() when initializing prevMeasure would fix this, but I would worry other problems might result.  I'd still love to see a better approach to the basic problem of the underlying measures never having been laid out, which has caused quite a few of these sorts of problems over the years.